File: src/FoodStuffs.Web/ClientApp/src/models/api/recipes/ListRecipesRequest.js
import { paginationTakeOptions } from '../../options';

export default class ListRecipesRequest {
  constructor() {
    this.category = '';
    this.name = '';
    this.sortBy = '';
    this.sortDesc = false;
    this.isPagingEnabled = true;
    this.page = 1;
    this.take = paginationTakeOptions[0].value;
  }
}

File: src/FoodStuffs.Model/Events/Recipes/ListRecipesRequest.cs
using VoidCore.Model.Responses.Collections;

namespace FoodStuffs.Model.Events.Recipes;

public record ListRecipesRequest(
    string? NameSearch,
    string? CategorySearch,
    string? SortBy,
    bool SortDesc,
    bool IsPagingEnabled,
    int Page,
    int Take) : IPaginatedRequest;
